### PR TITLE
Remove support for Kubernetes 1.8 and earlier

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 1]
```diff
@@ -1047,13 +1049,22 @@ func (c *ApplyClusterCmd) validateKubernetesVersion() error {
 	}
 
 	if !util.IsKubernetesGTE(OldestSupportedKubernetesVersion, *parsed) {
+		fmt.Printf("This version of Kubernetes is no longer supported; upgrading Kubernetes is required\n")
+		fmt.Printf("\n")
+		fmt.Printf("More information: %s\n", buildPermalink("upgrade_k8s", OldestRecommendedKubernetesVersion))
```
Comment (rifelpet) [inline upup/pkg/fi/cloudup/apply_cluster.go]: Any reason to use `OldestRecommendedKubernetesVersion` in the url anchor? There isn't an anchor for `#1.10.0` on the page currently.

Comment (johngmyers) [reply upup/pkg/fi/cloudup/apply_cluster.go]: I'm following the example from later in the function where it deprecates or removes support based on what is in the channel. It gives us the flexibility to add version-specific sections to the file at a later date.

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 2]
```diff
@@ -93,6 +93,8 @@ var (
 	CloudupModels = []string{"proto", "cloudup"}
 	// OldestSupportedKubernetesVersion is the oldest kubernetes version that is supported in Kops
 	OldestSupportedKubernetesVersion = "1.9.0"
+	// OldestRecommendedKubernetesVersion is the oldest kubernetes version that is not deprecated in Kops
+	OldestRecommendedKubernetesVersion = "1.10.0"
```
Comment (hakman) [inline upup/pkg/fi/cloudup/apply_cluster.go]: Maybe it's just me, but I would not recommend any k8s version where the default etcd is not 3.x. Anyone having to got through the upgrade from 2.x to 3.x will be pretty annoyed.
https://github.com/kubernetes/kops/blob/298f79659a2bd9c1d6548c6c95b335b5e29e6466/pkg/model/components/etcdmanager/options.go#L60-L61

Comment (johngmyers) [reply upup/pkg/fi/cloudup/apply_cluster.go]: The current policy as I understand it is to remove support from 1.8 and earlier in kops 1.18, then do rolling removals in subsequent versions. That means removing support from 1.9 in kops 1.19. There was generalized talk about narrowing the gap in future releases, but no specific decisions there have been made.

Comment (hakman) [reply upup/pkg/fi/cloudup/apply_cluster.go]: I don't have anything against supporting these versions, but the name says "recommended" version. I think "non-deprecated" is not quite the same as "recommended".

Comment (johngmyers) [reply upup/pkg/fi/cloudup/apply_cluster.go]: I am not able to find a better word for "non-deprecated". The code uses "recommended" for "non-deprecated" later in the file, where it's dealing with channels.

Comment (hakman) [reply upup/pkg/fi/cloudup/apply_cluster.go]: Maybe we can use "legacy" here.

Comment (johngmyers) [reply upup/pkg/fi/cloudup/apply_cluster.go]: The word "recommended" is not made visible to the user. The constant only demarcates what is not described as "deprecated".

**Comment by justinsb** [comment]
This LGTM - thanks @johngmyers 

Agree that we have a lot of version-thresholds now - deprecated, minimum-recommended, unsupported etc :-)   I imagine we'll figure this out over time - or maybe just align them!

/approve
/lgtm

File: upup/pkg/fi/cloudup/apply_cluster.go (hunk 2 above)
Comment (justinsb) [reply upup/pkg/fi/cloudup/apply_cluster.go]: Yeah, if & when we come up with better words we can rename this constant without affecting anything user-visible!

**Comment by justinsb** [comment]
Readding LGTM - thanks for fixing the staticcheck error @johngmyers 

/approve
/lgtm

**Pull request merged**